mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -442,53 +441,68 @@ def _parse_json_data(data, metadata, input_schema):
     return ParsedJsonInput(data, params, _is_unified_llm_input)
 
 
+def _async_catch_mlflow_exception(func):
+    from fastapi.responses import Response
+
+    @wraps(func)
+    async def wrapper(*args, **kwargs):
+        try:
+            return await func(*args, **kwargs)
+        except MlflowException as e:
+            return Response(
+                content=e.serialize_as_json(),
+                status_code=e.get_http_status_code(),
+                media_type="application/json",
+            )
+
+    return wrapper
+
+
 def init(model: PyFuncModel):
     """
     Initialize the server. Loads pyfunc model from the path.
     """
-    import flask
+    from fastapi import FastAPI, Request
+    from fastapi.responses import Response
 
-    app = flask.Flask(__name__)
+    app = FastAPI()
     input_schema = model.metadata.get_input_schema()
     # set the environment variable to indicate that we are in a serving environment
     os.environ[_MLFLOW_IS_IN_SERVING_ENVIRONMENT.name] = "true"
 
     @app.route("/ping", methods=["GET"])
     @app.route("/health", methods=["GET"])
-    def ping():
+    async def ping(request: Request):
         """
         Determine if the container is working and healthy.
         We declare it healthy if we can load the model successfully.
         """
         health = model is not None
         status = 200 if health else 404
-        return flask.Response(response="\n", status=status, mimetype="application/json")
+        return Response(content="\n", status_code=status, media_type="application/json")
 
     @app.route("/version", methods=["GET"])
-    def version():
+    async def version(request: Request):
         """
         Returns the current mlflow version.
         """
-        return flask.Response(response=VERSION, status=200, mimetype="application/json")
+        return Response(content=VERSION, status_code=200, media_type="application/json")
 
     @app.route("/invocations", methods=["POST"])
-    @catch_mlflow_exception
-    def transformation():
+    @_async_catch_mlflow_exception
+    async def transformation(request: Request):
```

Comment:
Is it possible to annotate the request payload with the modle signature?

Reply:
Not a blocker.

Reply:
This is a fantastic feature to add to populate the for-free REST UI interface via swagger UI. Agree that it is a nice-to-have followup.